routers/user/home.go

```diff
@@ -285,18 +285,18 @@ func Issues(ctx *context.Context) {
 			return
 		}
 
+		// Check if user has access to given repository.
+		if !repo.IsOwnedBy(ctxUser.ID) && !repo.HasAccess(ctxUser) {
+			ctx.Handle(403, "Issues", fmt.Errorf("#%d", repoID))
```

Comment:
Gogs does not expose repository as 403 if user does not have access, so we need to use `404`.